src/writer.rs

```diff
@@ -107,7 +107,7 @@ impl FdtWriter {
     /// # Arguments
     ///
     /// `mem_reservations` - reserved physical memory regions to list in the FDT header.
-    pub fn new(mem_reservations: &[FdtReserveEntry]) -> Self {
+    pub fn new(mem_reservations: &[FdtReserveEntry]) -> Result<Self> {
```

Comment:
I added a test based on the one in #6 and it seems to work (albeit very slowly). My main concern was that it will allocate 8 GiB of RAM at least (the original 4G array of FdtReserveEntry structures and a copy in the FdtWriter data), which could be a problem for smaller machines or tests running in a restricted envinronment; I guess we will see if this works on the CI system.